src/protocol/templates/ping_protocol.json

```diff
@@ -1,212 +1,511 @@
+      "undefined": {
+        "id": "0",
+        "payload": []
+      },
+      "ack": {
+        "id": "1",
+        "payload": []
+      },
+      "nack": {
+        "id": "2",
+        "payload": []
+      },
+      "ascii_text": {
+        "id": "3",
+        "payload": []
+      },
+      "set_device_id": {
+        "id": "1000",
+        "payload": [
+          {
+            "name": "device_id",
+            "type": "u8"
+          }
+        ]
+      },
+      "set_range": {
+        "id": "1001",
+        "payload": [
+          {
+            "name": "start_mm",
+            "type": "u8"
+          },
+          {
+            "name": "length_mm",
+            "type": "u8"
+          }
+        ]
+      },
+      "set_speed_of_sound": {
+        "id": "1002",
+        "payload": [
+          {
+            "name": "speed",
+            "type": "u32"
+          }
+        ]
+      },
+      "set_auto_manual": {
+        "id": "1003",
+        "payload": [
+          {
+            "name": "mode",
+            "type": "u8"
+          }
+        ]
+      },
+      "set_ping_rate_msec": {
+        "id": "1004",
+        "payload": [
+          {
+            "name": "rate_msec",
+            "type": "u16"
+          }
+        ]
+      },
+      "set_gain_index": {
+        "id": "1005",
+        "payload": [
+          {
+            "name": "index",
+            "type": "u8"
+          }
+        ]
+      },
+      "set_ping_enable": {
+        "id": "1006",
+        "payload": [
+          {
+            "name": "enable",
+            "type": "u8"
+          }
+        ]
+      },
+      "goto_bootloader": {
+        "id": "1100",
+        "payload": []
+      },
+      "fw_version": {
+        "id": "1200",
+        "payload": [
+          {
+            "name": "device_type",
+            "type": "u8"
+          },
+          {
+            "name": "device_model",
+            "type": "u8"
+          },
+          {
+            "name": "fw_version_major",
+            "type": "u16"
+          },
+          {
+            "name": "fw_version_minor",
+            "type": "u16"
+          }
+        ]
+      },
+      "device_id": {
+        "id": "1201",
+        "payload": [],
+        "deprecated": "true"
+      },
+      "voltage_5": {
+        "id": "1202",
+        "payload": [
+          {
+            "name": "mvolts",
+            "type": "u16"
+          }
+        ]
+      },
+      "speed_of_sound": {
+        "id": "1203",
+        "payload": [
+          {
+            "name": "speed_mmps",
+            "type": "u32"
+          }
+        ]
+      },
+      "range": {
+        "id": "1204",
+        "payload": [
+          {
+            "name": "start_mm",
+            "type": "u32"
+          },
+          {
+            "name": "length_mm",
+            "type": "u32"
+          }
+        ]
+      },
+      "mode": {
+        "id": "1205",
+        "payload": [
+          {
+            "name": "auto_manual",
```

Comment:
is this working ? does not appears to be here.

Reply:
I think this was never working right. We will have to do more testing and open a new issue.